lib/diagnostic.js

```diff
@@ -13,14 +19,33 @@ const min = function min() {
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let result = 0;
+  if (arguments.length > 0){
+    for (let i = 0; i < arguments.length; i++) {
+      result += arguments[i];
+    }
+  return result;
+}else{
+  let result = 0;
+  return result;
+}
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let result = [];
+  if (typeof defaultValue === "function"){
+    for (let i = 0; i < length; i++){
+      result[i] = defaultValue();
+    }
+  }else{
+    for(let i = 0; i < length; i++){
+      result[i] = defaultValue;
+    }
+  }
+  return result;
```

Comment:
If we want to set result to an array ( [ ] ), then we would need to use `result.push(defaultValue())` to add to that array.  If we are going to use `result[i] = defaultValue();` in the if/else then we need to write `let result = new Array(length);`  Something like this:
```
let array = new Array(length);
for (let i = 0; i < length; i++) {
  array[i];
  if (typeof defaultValue === 'function') {
	   array[i] = defaultValue(i);
   } else {
	    array[i] = defaultValue;
    }
}
return array;
```